### PR TITLE
Adds the log tag cqint for internal requests

### PR DESCRIPTION
**Pull request by zwoop**

### CODE REVIEW

**Comment by zwoop** [comment]
Testing on Docs now.

**Comment by randall** [comment]
[approve ci autest]

File: proxy/logging/LogAccess.cc [hunk 1]
```diff
@@ -1641,41 +1641,42 @@ int
...
   }
   return INK_MIN_ALIGN;
 }
 
+int
+LogAccess::marshal_client_req_is_internal(char *buf)
+{
+  if (buf) {
+    int64_t is_internal = m_http_sm->is_internal ? 1 : 0;
+    marshal_int(buf, is_internal);
```
Comment (ywkaras) [inline proxy/logging/LogAccess.cc]: Why not just:
```
marshal_int_buf, m_http_sm->is_internal ? 1 : 0);
```

File: proxy/http/HttpSM.cc [hunk 2]
```diff
@@ -464,6 +464,7 @@ HttpSM::attach_client_session(ProxyClientTransaction *client_vc, IOBufferReader
       _client_connection_id = p->connection_id();
     }
   }
+  is_internal = ua_txn->get_netvc()->get_is_internal_request();
```
Comment (ywkaras) [inline proxy/http/HttpSM.cc]: How about:
```
public:
  // Save is is_internal for later logging when UA transaction object no longer exists.
  void save_is_internal() { _is_internal = ua_txn->get_netvc()->get_is_internal_request(); }

  bool is_internal() const { return _is_internal; }

private:
  bool _is_internal;
```

Comment (SolidWallOfCode) [reply proxy/http/HttpSM.cc]: At this point I'd think it better to have a "history" subclass that is responsible for gathering this historical information for later use.

**Pull request merged**